### PR TITLE
Lolin S2 Mini - remove NeoPixel, pin assignments to match Wemos

### PR DESCRIPTION
**Pull request by durapensa**
Wemos/Lolin's MicroPython pin assignments for this board, plus pin assignments from Wemos D1 Mini board (S2 Mini is supposed to retain some shield compatibility with D1 Mini). Also, ESP32-S2 has hardware LED_PWM.

### CODE REVIEW

File: ports/esp32s2/boards/lolin_s2_mini/mpconfigboard.h [hunk 1]
```diff
@@ -27,21 +27,21 @@
 // Micropython setup
 
 #define MICROPY_HW_BOARD_NAME       "S2Mini"
-#define MICROPY_HW_MCU_NAME         "ESP32S2"
+#define MICROPY_HW_MCU_NAME         "ESP32S2-S2FN4R2" // from Wemos MP
 
-#define MICROPY_HW_NEOPIXEL (&pin_GPIO1)
-#define CIRCUITPY_STATUS_LED_POWER (&pin_GPIO15)
+// #define MICROPY_HW_NEOPIXEL (&pin_GPIO1) // no NeoPixel on S2 Mini
+// #define CIRCUITPY_STATUS_LED_POWER (&pin_GPIO15)
```
Comment (tannewt) [inline ports/esp32s2/boards/lolin_s2_mini/mpconfigboard.h:33]:
```suggestion
```

File: ports/esp32s2/boards/lolin_s2_mini/mpconfigboard.h [hunk 2]
```diff
@@ -27,21 +27,21 @@
...
-#define DEFAULT_SPI_BUS_MOSI (&pin_GPIO35)
-#define DEFAULT_SPI_BUS_MISO (&pin_GPIO36)
+#define DEFAULT_SPI_BUS_SCK (&pin_GPIO7)   // no SPI labels on S2 Mini, def from Wemos MP
+#define DEFAULT_SPI_BUS_MOSI (&pin_GPIO11) // no SPI labels on S2 Mini, def from Wemos MP
+#define DEFAULT_SPI_BUS_MISO (&pin_GPIO9)  // no SPI labels on S2 Mini, def from Wemos MP
 
-#define DEFAULT_UART_BUS_RX (&pin_GPIO44)
-#define DEFAULT_UART_BUS_TX (&pin_GPIO43)
+// #define DEFAULT_UART_BUS_RX (&pin_GPIO44) // no UART pins on S2 Mini
+// #define DEFAULT_UART_BUS_TX (&pin_GPIO43) // no UART pins on S2 Mini
```
Comment (tannewt) [inline ports/esp32s2/boards/lolin_s2_mini/mpconfigboard.h:47]:
```suggestion
```

File: ports/esp32s2/boards/lolin_s2_mini/mpconfigboard.mk [hunk 3]
```diff
@@ -14,7 +14,7 @@ CIRCUITPY_ESP_FLASH_MODE=qio
 CIRCUITPY_ESP_FLASH_FREQ=80m
 CIRCUITPY_ESP_FLASH_SIZE=4MB
 
-CIRCUITPY_BITBANG_NEOPIXEL = 1
+# CIRCUITPY_BITBANG_NEOPIXEL = 1 # no NeoPixel on S2 Mini
```
Comment (tannewt) [inline ports/esp32s2/boards/lolin_s2_mini/mpconfigboard.mk:17]:
```suggestion
```

**Review by tannewt — CHANGES_REQUESTED** [review]
Thanks for these updates. Just added suggestions to delete the things you commented out. No need to have them since they'll never be added.

**Comment by microDev1** [comment]
Reverted a commit adding `board_id`, since it is now set using a macro.

**Review by tannewt — APPROVED** [review]
Thanks for the updates! Looks good to me.

**Pull request merged**